pkg/storage/impl/mysql/mysql_impl.go

```diff
@@ -37,12 +63,103 @@ func (impl *MySQLImpl) GetConnectionType() taxonomy.ConnectionType {
 	return impl.Name
 }
 
-// storage allocation - placeholder
+func dsn(username, password, hostname, port, dbName string) string {
+	return fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", username, password, hostname, port, dbName)
+}
+
+// returns a connection to MySQL
+func NewClient(host, port, dbName string, secretRef taxonomy.SecretRef, client kclient.Client) (*sql.DB, error) {
+	// Get credentials
+	secret := v1.Secret{}
+	if err := client.Get(context.Background(), types.NamespacedName{Name: secretRef.Name,
+		Namespace: secretRef.Namespace}, &secret); err != nil {
+		return nil, errors.Wrapf(err, "could not get a secret %s", secretRef.Name)
+	}
+
+	username, password := string(secret.Data[usernameKey]), string(secret.Data[passwordKey])
+	if username == "" || password == "" {
+		return nil, errors.Errorf("could not retrieve credentials from the secret %s", secretRef.Name)
+	}
+
+	// connect to the server
+	return sql.Open(mysqlAgent, dsn(username, password, host, port, dbName))
+}
+
+// storage allocation
+// host, port are taken from the storage account
+// database name is generated, table is taken from the DatasetProperties.Name
+// TODO: allow database and table specification inside FybrikApplication
 func (impl *MySQLImpl) AllocateStorage(request *storagemanager.AllocateStorageRequest, client kclient.Client) (taxonomy.Connection, error) {
-	return taxonomy.Connection{Name: impl.Name}, nil
+	var host, port string
+	var err error
+	if host, err = agent.GetProperty(request.AccountProperties.Items, impl.Name, hostKey); err != nil {
+		return taxonomy.Connection{}, err
+	}
+	if port, err = agent.GetProperty(request.AccountProperties.Items, impl.Name, portKey); err != nil {
+		return taxonomy.Connection{}, err
+	}
+	portVal, err := strconv.Atoi(port)
+	if err != nil {
+		return taxonomy.Connection{}, err
+	}
+
+	// generate database name
+	suffix, _ := random.Hex(randomSuffixLength)
+	database := request.Opts.AppDetails.Name + "-" + request.Opts.AppDetails.Namespace + suffix
+
+	table := request.Opts.DatasetProperties.Name
+	// connect to the server and create the database if empty
```

Comment:
here there is a contradiction, if database name is auto generated, it should not exist, if we want to reuse the data base, we should not generate the suffix.

Reply:
It's for the future when database reuse becomes supported.